aggsender/types/types_test.go

```diff
@@ -249,3 +249,109 @@ func TestCertificateHeader_ElapsedTimeSinceCreation(t *testing.T) {
 		require.GreaterOrEqual(t, int64(dur.Seconds()), int64(10))
 	})
 }
+func TestAggsenderMode_Scan(t *testing.T) {
+	tests := []struct {
+		input    interface{}
+		expected AggsenderMode
+		hasError bool
+	}{
+		{"PessimisticProof", AggsenderMode("PessimisticProof"), false},
+		{"AggchainProof", AggsenderMode("AggchainProof"), false},
+		{"Auto", AggsenderMode("Auto"), false},
+		{"invalid", AggsenderMode(""), true},
+		{123, AggsenderMode(""), true}, // Non-string input
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("AggsenderMode_Scan_%v", tt.input), func(t *testing.T) {
+			var mode AggsenderMode
+			err := mode.Scan(tt.input)
+			if tt.hasError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tt.expected, mode)
+			}
+		})
+	}
+}
```

Comment:
Consider:
- relying upon checking the error message
- adding blank lines between tests (nitpick)
```suggestion

func TestAggsenderMode_Scan(t *testing.T) {
	tests := []struct {
		input       interface{}
		expected    AggsenderMode
		expectedErr string
	}{
		{"PessimisticProof", AggsenderMode("PessimisticProof"), ""},
		{"AggchainProof", AggsenderMode("AggchainProof"), ""},
		{"Auto", AggsenderMode("Auto"), ""},
		{"invalid", AggsenderMode(""), "AggsenderMode.Scan(...):"},
		{123, AggsenderMode(""), "AggsenderMode: expected string, got"}, // Non-string input
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("AggsenderMode_Scan_%v", tt.input), func(t *testing.T) {
			var mode AggsenderMode
			err := mode.Scan(tt.input)
			if tt.expectedErr != "" {
				require.ErrorContains(t, err, tt.expectedErr)
			} else {
				require.NoError(t, err)
				require.Equal(t, tt.expected, mode)
			}
		})
	}
}

```